python/cuml/preprocessing/model_selection.py

```diff
@@ -45,26 +46,33 @@ def _stratify_split(X, y, n_train, n_test, x_numba, y_numba, random_state):
     """
     x_cudf = False
     y_cudf = False
+
     if isinstance(X, cudf.DataFrame):
         x_cudf = True
-        X = X.values
     elif hasattr(X, "__cuda_array_interface__"):
         X = cp.asarray(X)
+        x_order = _strides_to_order(X.__cuda_array_interface__['strides'],
+                                    cp.dtype(X.dtype))
+
```

Comment:
Just a thought. Could some of this instance checking, order checking, branching logic, and (eventually) output type manipulation be avoided by using the CumlArray and storing the input types for the output?

I wonder if some of this code around instance and stride checking, branching with flag variables, and output type manipulation might be able to use leverage it to reduce the direct code complexity in this function.

Reply:
That's definitely possible. I followed this convention because non-stratified split doesn't make use of CumlArray. I think it might be worth making another PR to change the entire `train_test_split` call to make use of CumlArray

Reply:
Can you file an issue as a reminder to update this with CumlArray? Agreed that it could be helpful

Reply:
Sorry I missed this comment - filed the issue here https://github.com/rapidsai/cuml/issues/2870